### PR TITLE
Remeasurement

### PR DESCRIPTION
**Pull request by ankitecd**
Changes in this pull request:
- Added re-measurement for SYNCED plant locations

### CODE REVIEW

**Comment by norbertschuler** [comment]
I am sorry, I can not currently review this code as the app crashes in my iOS 15.4. simulator due to Mapbox (it's a known issue only for this emulator version).

File: .github/workflows/macos_build_ios.yml [hunk 1]
```diff
@@ -14,171 +14,171 @@ jobs:
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-treemapper2-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-treemapper2-
+
+      - name: Install npm
+        run: |
+          npm install --legacy-peer-deps
```
Comment (norbertschuler) [inline .github/workflows/macos_build_ios.yml:85]: Please merge develop where I already added `--legacy-peer-deps` and omit all your reformatting changes of all the GitHub workflow files.

**Comment by norbertschuler** [comment]
- [ ] The "Remeasure" button is only available on the map and not if you open the details of a polygon and look at a sample tree there. Also the button might not fit any longer translation in another language:

Screenshot_20220415-123957

**Comment by norbertschuler** [comment]
- [ ] The dialog for enter new data does not show the old data to be compare with. It also allows to not enter any new data at all:

Screenshot_20220415-122622

**Comment by norbertschuler** [comment]
- [ ] Once entered values can not be edited any more beside there seems to be an edit icon?

Screenshot_20220415-122656

**Comment by norbertschuler** [comment]
- [ ] I found not possibility to delete wrongly entered remeasurements again. But you can do serveral for the same trees in a row and probably all are going to be send to the backend: 

Screenshot_20220415-122720

**Comment by norbertschuler** [comment]
- [ ] For reporting trees as "dead" if you have not entered any new height an diameter the warning about wrong ratio is still shown:

Screenshot_20220415-124025

**Comment by norbertschuler** [comment]
- [ ] I got some errors trying upload my remeasurements to the staging backend: [TreeMapper v1.0.8 Logs.log](https://github.com/Plant-for-the-Planet-org/treemapper/files/8495560/TreeMapper.v1.0.8.Logs.log)

**Comment by norbertschuler** [comment]
- [ ] In the remeasuring process the back arrow while taking a picture does not work, both on the screen to take a picture as also on the screen to review the taken picture.

**Comment by norbertschuler** [comment]
- [ ] On the final review screen for a remeasuring the (x) button does the same as the "Save" button. You cannot abort the remeasuring any more.

File: .github/workflows/macos_build_ios.yml (hunk 1 above)
Comment (norbertschuler) [reply .github/workflows/macos_build_ios.yml:85]: We can also keep your version.

**Comment by norbertschuler** [comment]
- [ ] After having uploaded some remeasurements the local data shown on the map are still the old values. But if you hit the remeasure button, the new values are shown in the "Review" screen. So I assume the map interface is just not updated.
- [ ] Also on the "Review" screen now the HID is missing and the height and diameter cannot be edited any more.

Simulator Screen Shot - iPhone 12 - 2022-05-07 at 12 04 08Simulator Screen Shot - iPhone 12 - 2022-05-07 at 12 08 06

**Comment by norbertschuler** [comment]
- [ ] The pending remeasurings does not seems to count to the global indicator showing if all uploads are done:

Simulator Screen Shot - iPhone 12 - 2022-05-07 at 12 14 39

**Comment by norbertschuler** [comment]
@tejassonar is you think this is feature complete, please mark the PR "ready for review".

**Comment by tejassonar** [comment]
> @tejassonar is you think this is feature complete, please mark the PR "ready for review".

Re-easurement is ready to test, but I have not fully implemented the functionality of frequency and intensity (Which can be changed in Manage Projects).

**Comment by tejassonar** [comment]
> * [ ]  The pending remeasurings does not seems to count to the global indicator showing if all uploads are done:
> 
> Simulator Screen Shot - iPhone 12 - 2022-05-07 at 12 14 39

We need to maintain the re-measurement upload count separately, but we have not designed it for that yet. This can go in a later release maybe.

File: package.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "treemapper",
-  "version": "1.0.8",
+  "version": "1.0.9",
```
Comment (norbertschuler) [inline package.json:3]: Increasing release versions should not be part of feature branches, but should rather be done in release branches.

**Comment by norbertschuler** [comment]
- [ ] Got this error today:
```
 ERROR  ReferenceError: Can't find variable: isDateInRange

This error is located at:
    in SingleTreeOverview (at SceneView.tsx:130)
```

**Comment by norbertschuler** [comment]
> ```
>  ERROR  ReferenceError: Can't find variable: isDateInRange
> 
> This error is located at:
>     in SingleTreeOverview (at SceneView.tsx:130)
> ```

@tejassonar please have a look if just commenting this unknown variable is solving this issue correctly.

**Comment by norbertschuler** [comment]
- [ ] After uploading the remeasurement, the change is visible in the preview on the map, but not in the detail screen nor in the tree inventory when looking at uploaded items:

IMG_0872
IMG_0874IMG_0873

**Comment by norbertschuler** [comment]
> * [ ]  After uploading the remeasurement, the change is visible in the preview on the map, but not in the detail screen nor in the tree inventory when looking at uploaded items:

I tested on production and the remeasurement was not stored in the backend, so that is probably the reason for not showing the new data. There was no error message uploading the remeasurement, but after logging out and login again, the changed data has disappeared.

**Comment by norbertschuler** [comment]
- [ ] Logging out and in again with an TPO account results into this crash:
```
 ERROR  Error: Accessing object of type Projects which has been invalidated or deleted

This error is located at:
    in ProjectAndSiteSelector (at MainScreen/index.tsx:303)

```

**Comment by tejassonar** [comment]
> > * [ ]   After uploading the remeasurement, the change is visible in the preview on the map, but not in the detail screen nor in the tree inventory when looking at uploaded items:
> 
> I tested on production and the remeasurement was not stored in the backend, so that is probably the reason for not showing the new data. There was no error message uploading the remeasurement, but after logging out and login again, the changed data has disappeared on the map as well.
> 
> Here is my example request to save a remeasurement:
> 
> ```
> curl -v -X POST 'https://app-staging.plant-for-the-planet.org/treemapper/plantLocations/sloc_teIJFfnC1mY9Mka0Y46wk5Bg/event' -H 'accept: application/json, text/plain, */*' -H 'Accept-Encoding: gzip' -H 'authorization: Bearer ***' -H 'Connection: Keep-Alive' -H 'Content-Length: 103' -H 'Content-Type: application/json' -H 'Host: app-staging.plant-for-the-planet.org' -H 'user-agent: treemapper/android/1.0.9' -H 'x-session-id: lGbwZdAeIqXeYkxPWcrn2' -d '{"type":"measurement","eventDate":"2022-05-28T14:32:25.108Z","measurements":{"height":1.5,"width":2.5}}'
> 
> {"id":"sloc_teIJFfnC1mY9Mka0Y46wk5Bg","type":"sample","captureMode":"on-site","captureStatus":"complete","geometry":{"type":"Point","coordinates":[9.935373838444463,53.54868654796237]},"coordinates":[{"coordinateIndex":"0","image":"6292313f7c839678875349.jpg","status":"complete","id":"coord_mQ263ITk0Em0h25fYkhufvV8"}],"plantDate":"2022-05-28T00:00:00+00:00","plantedSpecies":[],"hid":"GNWF98","deviceLocation":{"type":"Point","coordinates":[9.9354217,53.5486067]},"originalGeometry":{"type":"Point","coordinates":[9.935373838444464,53.54868654796237]},"plantProject":null,"scientificSpecies":"sspec_i08DZUvNbyL0alNuEq","plantProjectSite":null,"image":null,"registrationDate":"2022-05-28T00:00:00+00:00","measurements":{"height":1.5,"width":2.5},"tag":null,"metadata":null,"sampleTreeCount":null,"parent":"loc_xcaeOKGIf6G6nJHXOi42061K"}
> ```

There are backend changes that are still required in order to implement this. Jorgo is going to work on that this week.  On the map, we see the data bcoz it's still present on the local. We don't get any measurement data from the backend yet.

File: app/actions/inventory.ts [hunk 3]
```diff
@@ -164,21 +178,21 @@ export const getAllInventoryFromServer = async (
...
+    //   // data.data.total,
+    //   // data.data.count,
+    //   // data.data._links,
+    //   // data.data._filters,
+    //   // '===',
+    //   // Object.keys(data.data),
+    //   // '++',
+    //   // Object.keys(data.data._links),
+    //   // Object.keys(data?.data?.items),
+    // );
```
Comment (sunilsabatp) [inline app/actions/inventory.ts:195]: remove console log if it is not needed any more

File: app/components/MainScreen/InventoryTypeSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+export const InventoryTypeSelector = () => {
+  const {state, dispatch} = useInventory();
+
+  const [showInventoryTypeDropdown, setShowInventoryTypeDropdown] = useState<boolean>(false);
+  const [selectedInvType, setSelectedInvType] = useState<InventoryType | null>(null);
+
+  const setInventoryType = async () => {
+    const user = await getUserDetails();
+    console.log(user);
```
Comment (sunilsabatp) [inline app/components/MainScreen/InventoryTypeSelector.tsx]: remove console log

File: app/components/MainScreen/InventoryTypeSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  }, []);
+
+  useEffect(() => {
+    switchInventoryFetchTypeFlag();
+  }, [selectedInvType]);
+
+  useEffect(() => {
+    console.log('selectedInvType', selectedInvType);
+  }, [selectedInvType]);
+
```
Comment (sunilsabatp) [inline app/components/MainScreen/InventoryTypeSelector.tsx]: Is this hook has any significance ?, if not then remove it

File: app/components/MainScreen/index.tsx [hunk 6]
```diff
@@ -325,6 +326,18 @@ export default function MainScreen() {
...
             </View>
+            {isUserLogin ? (
+              <View
+                style={{
+                  position: 'absolute',
+                  top: 75,
+                  left: 0,
+                  right: 25,
+                  width: '45%',
+                }}>
```
Comment (sunilsabatp) [inline app/components/MainScreen/index.tsx]: avoid inline styling

File: app/screens/ManageProjectsFlow/ManageProjects.tsx [hunk 7]
```diff
@@ -12,7 +12,7 @@ const ManageProjects = ({}: ManageProjectsProps) => {
   const navigation = useNavigation();
 
   const onProjectPress = (id: string) => {
-    navigation.navigate('ProjectConfig', { projectId: id });
+    // navigation.navigate('ProjectConfig', { projectId: id });
```
Comment (sunilsabatp) [inline app/screens/ManageProjectsFlow/ManageProjects.tsx:15]: remove this function if it is not having any significance

**Comment by norbertschuler** [comment]
- [ ] There is some partly hidden text next to the remeasure button (compare by [bug report in TestFlight](https://appstoreconnect.apple.com/apps/1524353784/testflight/screenshots/AEn5YS6ZO7Fq-WrlkWaJF3g?appPlatform=IOS&preReleaseVersionId=10d50b16-3705-4487-b9bf-991336aca733&sort=-timestamp)):

screenshot_01

**Pull request merged**